src/garrysmod/gamemodes/zombiesurvival/gamemode/player_movement/shared/player_movement.lua

```diff
@@ -39,16 +39,6 @@ function GM:Move(pl, move)
 			-- Use 7, because friction will amount this to a velocity of 1 roughly.
 			phase = pt.NoGhosting and E_GetDTFloat(pl, DT_PLAYER_FLOAT_WIDELOAD) > curtime()
 			M_SetMaxClientSpeed(move, math_min(M_GetMaxClientSpeed(move), phase and 7 or (36 * (pt.BarricadePhaseSpeedMul or 1))))
-		elseif not pt.NoBWSpeedPenalty then
-			fw = M_GetForwardSpeed(move)
-			if fw < 0 then
-				sd = M_GetSideSpeed(move)
-				if sd < 0 then sd = -sd end
-
-				if sd > fw then
-					M_SetMaxClientSpeed(move, M_GetMaxClientSpeed(move) * (P_GetActiveWeapon(pl).IsMelee and 0.75 or 0.5))
-				end
-			end
```

Comment:
Here's an explanation of how the movement system in Garry's Mod works: https://wiki.facepunch.com/gmod/Game_Movement.

I'd be interested to see if Garry's Mod's [default](https://github.com/Facepunch/garrysmod/blob/8aec7a5daf07b7210c361d78d32f8c8d865bc63c/garrysmod/gamemodes/base/gamemode/shared.lua#L162-L194) [implementation](https://github.com/Facepunch/garrysmod/blob/0e3cc50e551c0ba2eb0eb7dcd5d3f53eda7bd32e/garrysmod/gamemodes/base/gamemode/player_class/player_default.lua#L82-L85) is missing anything from CSGO's [implementation](https://github.com/TheAlePower/Counter-Strike-Global-Offensive/blob/9a0534100cb98ffa1cf0c32e138f0e7971e910d3/cstrike15_src/game/server/player_command.cpp#L133).